src/tests/illumidesk/spawners/test_illumidesk_dockerspawner.py

```diff
@@ -1,98 +1,254 @@
 import os
 import pytest
 import types
+
 from dockerspawner.dockerspawner import DockerSpawner
-from illumidesk.spawners.spawner import IllumiDeskDockerSpawner
+
+from illumidesk.spawners.spawners import IllumiDeskBaseDockerSpawner
+from illumidesk.spawners.spawners import IllumiDeskRoleDockerSpawner
+from illumidesk.spawners.spawners import IllumiDeskWorkSpaceDockerSpawner
+
+
+def test_instantiate_base_illumidesk_docker_spawner_returns_not_implemented_error():
```

Comment:
I think this test is not needed because the method can be replaced (in base classes) and it's not common to test a base/empty method that is marked as `not implemented`